scripts/viz/tsne_viz.R

```diff
@@ -1,25 +1,47 @@
 # Pan-Cancer Variational Autoencoder
 # Gregory Way 2017
 #
-# Visualize t-SNE plots of VAE and raw RNAseq data features
+# Visualize t-SNE plots of VAE, ADAGE, and raw RNAseq data features
 
 library(dplyr)
 library(ggplot2)
 
+plot_tsne <- function(tsne_df) {
+  # Function to output tsne visualizations colored by cancer type for the given
+  # input data (compressed by either Tybalt or ADAGE or raw RNAseq data)
+  p <- ggplot(tsne_df, aes(x = `1`, y = `2`, color = acronym)) +
+    geom_point(size = 0.001) +
+    scale_colour_manual(limits = tcga_colors$`Study Abbreviation`,
+                        values = tcga_colors$`Hex Colors`,
+                        na.value = "black",
+                        breaks = palette_order) +
+    theme_classic() +
+    theme(legend.title = element_text(size = 8),
+          legend.text = element_text(size = 5),
+          legend.key.size = unit(0.5, "line")) +
+    guides(colour = guide_legend(override.aes = list(size = 0.1),
+                                 title = 'Cancer-Type'))
+  return(p)
+}
+
 # Load encodings file with matched clinical data to subset
 clinical_file <- file.path("data", "tybalt_features_with_clinical.tsv")
 clinical_df <- readr::read_tsv(clinical_file)
 clinical_df <- clinical_df %>% dplyr::select(sample_id, acronym)
 
 # Load tsne data
 tsne_vae_file <- file.path("results", "tybalt_tsne_features.tsv")
+tsne_adage_file <- file.path("results", "adage_tsne_features.tsv")
 tsne_rnaseq_file <- file.path("results", "rnaseq_tsne_features.tsv")
 
 tsne_vae_df <- readr::read_tsv(tsne_vae_file)
+tsne_adage_df <- readr::read_tsv(tsne_adage_file)
 tsne_rnaseq_df <- readr::read_tsv(tsne_rnaseq_file)
 
 tsne_vae_df <- dplyr::inner_join(tsne_vae_df, clinical_df,
                                  by = c("tcga_id" = "sample_id"))
+tsne_adage_df <- dplyr::inner_join(tsne_adage_df, clinical_df,
+                                 by = c("tcga_id" = "sample_id"))
 tsne_rnaseq_df <- dplyr::inner_join(tsne_rnaseq_df, clinical_df,
                                     by = c("tcga_id" = "sample_id"))
 
```

Comment:
Looks like you're applying the same commands several times, not sure if there is a "sleeker" way to do this in R like there is in python:

data_folder = 'C:/Users/alexj/Documents/UPenn/GCB537/Ex1prog/'
file_motifs = [data_folder + 'Output_motif%d_sequences_modified.txt' % (i+1) for i in xrange(n_data_files)]

Just a question